tests/shardddl3/run.sh

```diff
@@ -871,6 +871,1199 @@ function DM_112() {
+         (partition p0 values less than (100), partition p1 values less than (200));\"" \
+        "clean_table" "pessimistic"
+    run_case 144 "double-source-optimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id) \
+        (partition p0 values less than (100), partition p1 values less than (200));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (id int primary key) partition by range(id) \
+         (partition p0 values less than (100), partition p1 values less than (200));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (id int primary key) partition by range(id) \
+         (partition p0 values less than (100), partition p1 values less than (200));\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_145_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(100),(101),(102),(103),(104),(105);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(200),(201),(202),(203),(204),(205);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(300),(301),(302),(303),(304),(305);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} engine=innodb;"
+
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "query-status test" \
+        "This type of ALTER TABLE is currently unsupported" 1
+}
+
+# Defragment.
+function DM_145 {
+    run_case 145 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    run_case 145 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
+}
+
+function DM_146_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(100),(101),(102),(103),(104),(105);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(200),(201),(202),(203),(204),(205);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(300),(301),(302),(303),(304),(305);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} ROW_FORMAT=COMPACT;"
+
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "query-status test" \
+        "This type of ALTER TABLE is currently unsupported" 1
+}
+
+# Modify row format.
+function DM_146 {
+    run_case 146 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    run_case 146 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
+}
+
+function DM_147_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(100),(101),(102),(103),(104),(105);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(200),(201),(202),(203),(204),(205);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(300),(301),(302),(303),(304),(305);"
+
+    # FIXME: DM should detect table renaming and report an error.
+    run_sql_source1 "use ${shardddl1}; rename table ${tb1} to ${tb1}_new;"    
+
+    run_sql_source1 "use ${shardddl1}; drop table if exists ${tb1}_new;"
```

Comment:
ditto

Reply:
Test case for "rename table" depends on later development of rename detecting and error reporting. We can't do much at this moment.

Reply:
thank for waiting this PR https://github.com/pingcap/dm/pull/1512